akka-typed/src/main/scala/akka/typed/internal/ActorCell.scala

```diff
@@ -0,0 +1,433 @@
+      // this is not an off-by-one: #msgs is activations-1 if >0
+      if (oldActivations > maxQueue) {
+        if (Debug) println(s"[$thread] $self NOT enqueueing $msg at status $old ($oldActivations > $maxQueue)")
+        // cannot enqueue, need to give back activation token
+        unsafe.getAndAddInt(this, statusOffset, -1)
+        system.eventStream.publish(Dropped(msg, self))
+      } else if (ActorCell.isTerminating(old)) {
+        if (Debug) println(s"[$thread] $self NOT enqueueing $msg at status $old (is terminating)")
+        unsafe.getAndAddInt(this, statusOffset, -1)
+        system.deadLetters ! msg
+      } else {
+        if (Debug) println(s"[$thread] $self enqueueing $msg at status $old")
+        // need to enqueue; if the actor sees the token but not the message, it will reschedule
+        queue.add(msg)
+        if (oldActivations == 0) {
+          if (Debug) println(s"[$thread] $self being woken up")
+          unsafe.getAndAddInt(this, statusOffset, 1) // the first 1 was just the “active” bit, now add 1msg
+          // if the actor was not yet running, set it in motion; spurious wakeups don’t hurt
+          executionContext.execute(this)
+        }
+      }
+    } catch handleException
+
+  def sendSystem(signal: SystemMessage): Unit = {
+    @tailrec def needToActivate(): Boolean = {
+      val currentList = _systemQueue
+      if (currentList.head == NoMessage) {
+        system.deadLetters.toImpl.sendSystem(signal)
+        false
+      } else {
+        unsafe.compareAndSwapObject(this, systemQueueOffset, currentList.head, (signal :: currentList).head) || {
+          signal.unlink()
+          needToActivate()
+        }
+      }
+    }
+    try {
+      if (needToActivate()) {
+        val old = unsafe.getAndAddInt(this, statusOffset, 1)
+        if (isClosed(old)) {
+          // nothing to do
+          if (Debug) println(s"[$thread] $self NOT enqueueing $signal: terminating")
+        } else if (activations(old) == 0) {
+          // all is good: we signaled the transition to active
+          if (Debug) println(s"[$thread] $self enqueueing $signal: activating")
+          executionContext.execute(this)
+        } else {
+          // take back that token: we didn’t actually enqueue a normal message and the actor was already active
+          if (Debug) println(s"[$thread] $self enqueueing $signal: already active")
+          unsafe.getAndAddInt(this, statusOffset, -1)
+        }
+      } else if (Debug) println(s"[$thread] $self NOT enqueueing $signal: terminated")
+    } catch handleException
+  }
+
+  /**
+   * Main entry point into the actor: the ActorCell is a Runnable that is
+   * enqueued in its Executor whenever it needs to run. The _status field is
+   * used for coordination such that it is never enqueued more than once at
+   * any given time, because that would break the Actor Model.
+   *
+   * The idea here is to process at most as many messages as were in queued
```

Comment:
Because the constant would need to be tuned while this scheme has no tunable parameters. I wanted to start with the simplest thing, then measure and elaborate.